### PR TITLE
Add config for debugging individual test files

### PR DESCRIPTION
**Pull request by ckant**
## Description

Restricts the tests to the current opened file
https://code.visualstudio.com/docs/editor/variables-reference#_predefined-variables

## Motivation and Context

Easier to run individual tests

## Testing

Tests were tested

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: third-party/test/index.ts [hunk 1]
```diff
@@ -118,8 +118,12 @@ function run(testsRoot, clb): any {
         coverageRunner.setupCoverage();
     }
 
+    // 2020-03-24: Amazon addition.
+    const testFile = process.env["TEST_FILE"]?.replace(/^src\/test\//, "")?.concat('.js')
+    // END 2020-03-24: Amazon addition.
```
Comment (bryceitoc9) [inline third-party/test/index.ts]: Nit: move the Amazon addition to after the `glob`

File: .vscode/launch.json [hunk 2]
```diff
@@ -41,6 +41,22 @@
...
+            "type": "extensionHost",
+            "request": "launch",
+            "runtimeExecutable": "${execPath}",
+            "args": [
+                "--extensionDevelopmentPath=${workspaceFolder}",
+                "--extensionTestsPath=${workspaceFolder}/dist/src/test/index.js"
+            ],
+            "env": {
+                "TEST_REPORT_DIR": "${workspaceFolder}/.test-reports",
+                "TEST_FILE": "${relativeFileDirname}/${fileBasenameNoExtension}"
```
Comment (bryceitoc9) [inline .vscode/launch.json]: Just making sure: what happens if the current file isn't a test?

File: third-party/test/index.ts (hunk 1 above)
Comment (awschristou) [reply third-party/test/index.ts]: Good catch, this is more than a nit, since we should mark our divergent changes within the third-party folder

Comment (ckant) [reply third-party/test/index.ts]: Is there a reason we're doing this? MIT license doesn't require that you state your changes. It just requires that you include the MIT license.

Was this for our own purposes?

File: .vscode/launch.json (hunk 2 above)
Comment (ckant) [reply .vscode/launch.json]: The glob matches and runs 0 tests successfully 😄

**Pull request merged**